assignments/rustlatin.adoc

```diff
@@ -0,0 +1,146 @@
+= Exercise: rustlatin
+:source-language: rust
+
+In this exercise we will implement a Rust-y, simpler variant of https://en.wikipedia.org/wiki/Pig_Latin[Pig Latin]
+
+You will learn:
+
+* How to create a Rust library
+* How to deal with `Strings` and splitting
+* How to deal with `char` of a `String`
+* An early taste of Iterators
+* How to define Globals
+* Arrays vs Vectors
+* See Rust compiler's type inference in action
+* Most common way to do string concatenation
+
+== Specification
+
+For this exercise we define 
+
+* the Vowels of English alphabet -> `['a', 'e', 'i', 'o', 'u']`
+* a sentence is a collection of ASCII characters with words that are separated by a white space
+
+For any given sentence, you have to modify each word of the sentence using the following logic:
+
+* If the word begins with a vowel -> add prefix “sr” to the word
+* If the word does not begin with a vowel -> add suffix “rs” to the word
+
+== Tasks
+
+=== Step 1
+
+Create a new `lib` and name it `rustlatin`. 
+
+.Click to see the solution
+[%collapsible]
+====
+[source,bash]
+----
+cargo new --lib rustlatin
+----
+====
+
+=== Step 2
+
+Create a global variable in the file that defines the Vowels as specified above
+
+.Click to see the solution
+[%collapsible]
+====
+[source,rust]
+----
+const VOWELS: [char; 5] = ['a', 'e', 'i', 'o', 'u'];
+----
+====
+
+=== Step 3
+
+Copy paste this skeleton of the function
+
+[source,rust]
+----
+fn rustlatin(sentence: String) -> String {
+    unimplemented!()
+}
+----
+
+.Click to see the sample solution
+[%collapsible]
+====
+[source,rust]
+----
+fn rustlatin(sentence: String) -> String {
+    let words: Vec<_> = sentence.split(' ').collect();
+    let mut new_words = Vec::new();
+    for word in words {
```

Comment:
```suggestion
    for word in sentence.split(' ') {
```

I don't think there's a need for the intermediate collection.

Reply:
Ah I think I try to work in the editor so maybe a copy-pasta error. Apologies.